alcs-frontend/src/app/features/home/subtask/subtask.component.ts

```diff
@@ -22,14 +22,28 @@ export class SubtaskComponent implements OnInit {
   noticeOfIntentSubtasks: HomepageSubtaskDto[] = [];
   nonApplicationSubtasks: HomepageSubtaskDto[] = [];
 
-  constructor(private homeService: HomeService, private userService: UserService) {}
+  showNoi = true;
+  showAppAndNonApp = true;
+
+  constructor(
+    private homeService: HomeService,
+    private userService: UserService,
+    private authService: AuthenticationService
+  ) {}
 
   ngOnInit(): void {
     this.userService.$assignableUsers.subscribe((users) => {
       this.users = users.filter((user) => user.clientRoles.some((role) => this.assignableRoles.includes(role)));
     });
     this.userService.fetchAssignableUsers();
 
+    this.authService.$currentUser.subscribe((currentUser) => {
+      if (currentUser && this.subtaskType === CARD_SUBTASK_TYPE.PEER_REVIEW) {
+        this.showNoi = !!currentUser.client_roles && currentUser.client_roles.includes(ROLES.SOIL_OFFICER);
+        this.showAppAndNonApp = !!currentUser.client_roles && currentUser.client_roles.includes(ROLES.LUP);
+      }
+    });
```

Comment:
Does this need `takeUntil` to prevent memory leaks

Reply:
done